File: Weijia/121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        i=0
        c_min=prices[0]
        max_pro=0
        while i < len(prices):
            c_min=min(c_min,prices[i])
            curren_pro=prices[i]-c_min
            max_pro=max(max_pro,curren_pro)
            i+=1
        return max_pro

File: solution_zzj/121._Best_Time_to_Buy_and_Sell_Stock.cc
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0, min_buy = prices[0];
        for (int i=1;i<prices.size();++i) {
            ans = max(ans, prices[i]-min_buy);
            min_buy = min(min_buy, prices[i]);
        }
        return ans;
    }

	int maxProfit(vector<int>& prices) {
        int ans = 0, min_left = 0;
        for (int i=0;i<prices.size();++i) {
            if (prices[i]<prices[min_left]) {
                min_left = i;
            }
            ans = max(ans, prices[i]-prices[min_left]);
        }
        return ans;
    }
};
